### PR TITLE
DOC: explain EWM

### LINKED ISSUE #34867: DOC: pandas/pandas/core/window/ewm.py
#### Location of the documentation
Docstring for `class EWM(_Rolling)` in ewm.py.

#### Documentation problem
The purposes of the `EWM` class is unclear because there's no documentation on what 'ewm' is abbreviating. One reason this is confusing is that `ewm` can easily be interpreted as either "exponentially weighted mean" or "exponentially weighted moving". The `EWM` class docstring actually muddies the water further by stating that the class provides "exponential weighted **(EW)** functions".

#### Suggested fix for documentation
If it's absolutely necessary to use an abbreviation like `ewm` then the docstring needs to clearly state what it's abbreviating.

### PR DESCRIPTION
**Pull request by KrishnaSai2020**
- [ ] closes #34867
- [ ] tests  passed
- [ ] passes `black pandas`
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew - Made docs more clear and renamed files

### CODE REVIEW

**Comment by KrishnaSai2020** [comment]
> Hello @KrishnaSai2020! Thanks for opening this PR. We checked the lines you've touched for [PEP 8](https://www.python.org/dev/peps/pep-0008) issues, and found:
> 
> * In the file [`pandas/core/window/ewf.py`](https://github.com/pandas-dev/pandas/blob/fb5eac01beef3b60f526d02d608175779604b899/pandas/core/window/ewf.py):
> 
> > [Line 66:89](https://github.com/pandas-dev/pandas/blob/fb5eac01beef3b60f526d02d608175779604b899/pandas/core/window/ewf.py#L66): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (102 > 88 characters)

Yeah this pep8 conflict was added by previous contributors not done by me

**Comment by KrishnaSai2020** [comment]
right I'll fix that pep 8 issue and reopen this pr

**Comment by charlesdong1991** [comment]
thanks for the PR @KrishnaSai2020 

can you pls fix it though? Not sure about what the previous contributor did, but unlikely his/her PR got merged with the PEP8 issue. 

Looks like your expanded `EM` to `Exponential Weighted` caused it.

**Comment by KrishnaSai2020** [comment]
Ah I see! should i close this, fix and reopen or is there an easier way to edit the PR??

**Comment by charlesdong1991** [comment]
you can fix it in this PR, no need to close it and reopen a new one, thanks!

File: pandas/core/generic.py [hunk 1]
```diff
@@ -10460,7 +10460,7 @@ def _add_series_or_dataframe_operations(cls):
         Add the series or dataframe only operations to the cls; evaluate
         the doc strings again.
         """
-        from pandas.core.window import EWM, Expanding, Rolling, Window
+        from pandas.core.window import EWF, Expanding, Rolling, Window
```
Comment (jreback) [inline pandas/core/generic.py:10463]: i would be ok with actually writing this classname out if we are going to change it, e.g. ExponentialMoving or ExponentialMovingWindow

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/generic.py (hunk 1 above)
Comment (KrishnaSai2020) [reply pandas/core/generic.py:10463]: sounds good might as well

Comment (KrishnaSai2020) [reply pandas/core/generic.py:10463]: quick one could you clarify what you mean by writing it out?

**Comment by KrishnaSai2020** [comment]
@charlesdong1991  fixed the pep 8 problem

File: pandas/core/generic.py (hunk 1 above)
Comment (charlesdong1991) [reply pandas/core/generic.py:10463]: I _think_ @jreback meant you could rename `EWM` to `ExponentialMoving` in `core.window.ewm.py` where the class is created

Comment (KrishnaSai2020) [reply pandas/core/generic.py:10463]: k

File: pandas/core/window/ewf.py [hunk 2]
```diff
@@ -59,11 +59,11 @@ def get_center_of_mass(
     return float(comass)
 
 
-class EWM(_Rolling):
+class Exponential_moving(_Rolling):
```
Comment (charlesdong1991) [inline pandas/core/window/ewf.py:62]: could you use `ExpontentialMoving` other than `Expontential_moving`?

**Review by KrishnaSai2020 — COMMENTED** [review]
please re-review

File: pandas/core/generic.py (hunk 1 above)
Comment (KrishnaSai2020) [reply pandas/core/generic.py:10463]: changes resolved

File: pandas/core/window/ewf.py (hunk 2 above)
Comment (KrishnaSai2020) [reply pandas/core/window/ewf.py:62]: sure

**Comment by charlesdong1991** [comment]
I _think_ what was meant by @jreback is to rename `EWM` to `ExponentialMoving` or `ExpontentialMovingWindow` other than to `EWF`.

so you might also need to replace all your `EWF` to the new class name you choose.

**Comment by KrishnaSai2020** [comment]
@charlesdong1991 I think that's all instances refactored let me know if there more.

File: pandas/core/generic.py [hunk 3]
```diff
@@ -10460,7 +10460,8 @@ def _add_series_or_dataframe_operations(cls):
         Add the series or dataframe only operations to the cls; evaluate
         the doc strings again.
         """
-        from pandas.core.window import EWM, Expanding, Rolling, Window
+        from pandas.core.window import exponentialmoving as em
```
Comment (charlesdong1991) [inline pandas/core/generic.py:10463]: i don't think is necessary now to rename the file

File: pandas/core/generic.py [hunk 4]
```diff
@@ -10520,7 +10521,7 @@ def ewm(
             axis=0,
         ):
             axis = self._get_axis_number(axis)
-            return EWM(
+            return EWF(
```
Comment (charlesdong1991) [inline pandas/core/generic.py:10524]: still need to change?

File: pandas/core/generic.py [hunk 5]
```diff
@@ -10532,7 +10533,7 @@ def ewm(
                 axis=axis,
             )
 
-        cls.ewm = ewm
+        cls.eem = em
```
Comment (charlesdong1991) [inline pandas/core/generic.py:10536]: emm, why the change here?

File: pandas/core/window/__init__.py [hunk 6]
```diff
@@ -1,3 +1,3 @@
-from pandas.core.window.ewm import EWM  # noqa:F401
+from pandas.core.window.ewf import exponentialmoving   # noqa:F401
```
Comment (charlesdong1991) [inline pandas/core/window/__init__.py:1]: is there still `ewf` file?

File: pandas/core/window/exponentialmoving.py [hunk 7]
```diff
@@ -59,11 +59,11 @@ def get_center_of_mass(
     return float(comass)
 
 
-class EWM(_Rolling):
+class Exponentialmoving(_Rolling):
```
Comment (charlesdong1991) [inline pandas/core/window/exponentialmoving.py:62]: i think they might prefer capital letter for `Moving`

File: pandas/core/window/exponentialmoving.py [hunk 8]
```diff
@@ -185,7 +185,7 @@ def __init__(
 
     @property
     def _constructor(self):
-        return EWM
+        return EWF
```
Comment (charlesdong1991) [inline pandas/core/window/exponentialmoving.py:188]: this still need to change too?

**Review by charlesdong1991 — COMMENTED** [review]
thanks, several comments

File: pandas/core/generic.py (hunk 3 above)
Comment (jreback) [inline pandas/core/generic.py:10463]: use python naming conventions, e.g. ExponentialMoving, and don't abbreviate as this is a class

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/window/exponentialmoving.py (hunk 8 above)
Comment (KrishnaSai2020) [reply pandas/core/window/exponentialmoving.py:188]: Tell you what mate give me some time I'll rollback all the changes, start afresh and create a new pull request. i've confused myself. thanks for your help!

File: pandas/core/generic.py (hunk 3 above)
Comment (KrishnaSai2020) [reply pandas/core/generic.py:10463]: k thanks i'll keep that in mind

**Comment by KrishnaSai2020** [comment]
https://github.com/pandas-dev/pandas/pull/34910
